odl/solvers/functional/functional.py

```diff
@@ -822,6 +823,83 @@ def __str__(self):
                                           self.translation)
 
 
+class InfimalConvolution(Functional):
+
+    """The functional representing ``h(x) = inf_y f(x-y) + g(y)``."""
+
+    def __init__(self, funcA, funcB):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        funcA : `Functional`
+            Function corresponding to ``f``.
+        funcB : `Functional`
+            Function corresponding to ``g``.
+
+        Example
+        -------
```

Comment:
A short line describing what's happening here would be great.

A tip to make it easier to understand would be to create functionals that are themselves convex conjugates, that way evaluating the convex conjugate of this will give an "obvious" answer.